mirrord-layer/src/pod_api.rs

```diff
@@ -162,6 +163,14 @@ pub async fn create_agent(
 
     let _ = tokio::time::timeout(std::time::Duration::from_secs(20), running)
         .await
-        .unwrap();
-    pods_api.portforward(&pod_name, &[61337]).await.unwrap()
+        .with_context(|| {
```

Comment:
And here you change this `with_context` to `expect`(same for every similar piece of code).

Reply:
I don't think we should call expect from this part of the code.. rather return an error and the top function should decide whether to panic (in this case, it should because it's a ctor function and can't do anything else but panic)

Reply:
Hey Dan, talked with aviram, and we decided to go for the `Result` approach (sorry about potential confusion).

Reply:
Haha it's all good, friends. Just to make sure I'm up to date - are there any changes you need before we merge this and I get started on #62 ? I updated the branch to be in line with `upstream/main`, so y'all just let me know if anything needs changing :)